runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/util/MemoryMonitor.java

```diff
@@ -588,6 +639,61 @@ public File dumpHeap()
     return dumpHeap(localDumpFolder);
   }
 
+  /**
+   * Run a JFR profile, returning the profile data as an InputStream when complete.
+   *
+   * @param duration The amount of time to record for.
+   */
+  public CompletableFuture<InputStream> runJfrProfile(Duration duration) {
+    if (jfrInterop == null) {
+      throw new IllegalStateException("unable to run JFR profile because JFR was not enabled");
```

Comment:
```suggestion
      throw new IllegalStateException("Unable to run JFR profile because JFR was not enabled. You must use Java 9+ to enable JFR.");
```